src/scantailor/core/zones.py

```
@@ -0,0 +1,285 @@
+class ZoneSet(BaseModel):
+    """A collection of zones.
+
+    Supports iteration and provides methods for adding/removing zones.
+    """
+
+    model_config = {"arbitrary_types_allowed": True}
+
+    _picture_zones: list[Zone] = []
+    _fill_zones: list[Zone] = []
+
+    def __init__(self, **data) -> None:
```

Ruff (0.15.0)

[warning] Missing type annotation for `**data`

(ANN003)